src/echo/ws-api/wallet-api.js

```diff
@@ -260,249 +253,287 @@ class WalletAPI {
 	 */
-	normalizeBrainKey(brainKey) {
-		return this.wsRpc.call([0, 'normalize_brain_key', [string.toRaw(brainKey)]]);
-	}
+	normalizeBrainKey(brainKey) { return this.wsRpc.call([0, 'normalize_brain_key', [string.toRaw(brainKey)]]); }
 
 	/**
-	 * @method saveWalletFile
-	 * @param {String} walletFilename
+	 * Saves the current wallet to the given filename.
+	 *
+	 * **WARNING:** This does not change the wallet filename that will be used for future writes,
+	 * so think of this function as 'Save a Copy As...' instead of 'Save As...'.
+	 * Use `setWalletFilename` method to make the filename persist.
+	 * @param {string} walletFilename the filename of the new wallet JSON file to create or overwrite.
+	 * If `wallet_filename` is empty, save to the current filename
 	 * @returns {Promise<void>}
 	 */
-	saveWalletFile(walletFilename) {
-		return this.wsRpc.call([0, 'save_wallet_file', [string.toRaw(walletFilename)]]);
-	}
+	saveWalletFile(walletFilename) { return this.wsRpc.call([0, 'save_wallet_file', [string.toRaw(walletFilename)]]); }
 
 	/**
-	 * @method listMyAccounts
-	 *
-	 * @returns {Promise<Object[]>}
+	 * Lists all accounts controlled by this wallet.
+	 * This returns a list of the full account objects for all accounts whose private keys we possess.
+	 * @returns {Promise<any[]>} a list of account objects
 	 */
-	listMyAccounts() {
-		return this.wsRpc.call([0, 'list_my_accounts', []]);
-	}
+	listMyAccounts() { return this.wsRpc.call([0, 'list_my_accounts', []]); }
 
 	/**
-	 * @method listAccounts
-	 * @param {String} accountName
-	 * @param {Number} limit
-	 * @returns {Promise<[string, string][]>}
+	 * Lists all accounts registered in the blockchain.
+	 * This returns a list of all account names and their account ids, sorted by account name.
+	 * Use the `lowerbound` and limit parameters to page through the list. To retrieve all accounts,
+	 * start by setting `lowerbound` to the empty string `""`, and then each iteration, pass
+	 * the last account name returned as the `lowerbound` for the next `listAccounts` call.
+	 * @param {string} lowerbound the name of the first account to return. If the named account does not exist,
+	 * the list will start at the account that comes after `lowerbound`
+	 * @param {number} limit limit the maximum number of accounts to return (max: 1000)
+	 * @returns {Promise<[string, string][]>} a list of accounts mapping account names to account ids
 	 */
-	listAccounts(accountName, limit = API_CONFIG.LIST_ACCOUNTS_DEFAULT_LIMIT) {
-		if (limit > API_CONFIG.LIST_ACCOUNTS_MAX_LIMIT) {
-			throw new Error(`field 'limit' should be less than or equals to ${API_CONFIG.LIST_ACCOUNTS_MAX_LIMIT}`);
+	listAccounts(lowerbound, limit = API_CONFIG.LIST_ACCOUNTS_DEFAULT_LIMIT) {
+		if (!isAccountName(lowerbound)) throw new Error('Account name should be string and valid');
+		if (!limit > API_CONFIG.LIST_ACCOUNTS_MAX_LIMIT) {
+			throw new Error(`Limit should be capped at ${API_CONFIG.LIST_ACCOUNTS_MAX_LIMIT}`);
```

Comment:
This function (like another in this class) returns `Promise`. Use `Promise.reject` to avoid errors when `.catch` used for errors handling.